### PR TITLE
Testfix/appium

### PR DESCRIPTION
**Pull request by Hyphen31**
Updates to allow frameworkium-core to work with the new Microsoft WinAppDriver via Appium.

### CODE REVIEW

**Comment by jleeh** [comment]
Can this branch be rebased and commits squashed as much as possible?

File: src/test/java/com/frameworkium/integration/angularjs/pages/web/DeveloperGuidePage.java [hunk 1]
```diff
@@ -44,7 +44,6 @@ public DeveloperGuidePage searchDeveloperGuide(String inputText) {
 
     public DeveloperGuidePage clickBootstrapSearchItem() {
         bootstrapSearchItem.click();
-        waitForJavascriptFrameworkToFinish();
```
Comment (jleeh) [inline src/test/java/com/frameworkium/integration/angularjs/pages/web/DeveloperGuidePage.java]: Does this have to be removed? Does the native app run this test?

File: src/test/java/com/frameworkium/integration/angularjs/tests/web/DocumentationTest.java [hunk 2]
```diff
@@ -11,12 +11,6 @@
 
 public class DocumentationTest extends BaseTest {
 
-    @BeforeMethod
-    public void test_configure_browser_before_use() {
```
Comment (jleeh) [inline src/test/java/com/frameworkium/integration/angularjs/tests/web/DocumentationTest.java]: Same for this.

File: src/main/java/com/frameworkium/core/ui/driver/AbstractDriver.java [hunk 3]
```diff
@@ -62,7 +62,9 @@ private WebDriverWrapper setupEventFiringWebDriver(DesiredCapabilities caps) {
         if (ScreenshotCapture.isRequired()) {
             eventFiringWD.register(new CaptureListener());
         }
-        eventFiringWD.manage().timeouts().setScriptTimeout(DEFAULT_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+        if (!Driver.isNative()) {
+        	eventFiringWD.manage().timeouts().setScriptTimeout(DEFAULT_TIMEOUT_SECONDS, TimeUnit.SECONDS);
```
Comment (ham1) [inline src/main/java/com/frameworkium/core/ui/driver/AbstractDriver.java]: spaces not tabs please

File: src/main/java/com/frameworkium/core/ui/driver/drivers/WinAppDriverImpl.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import org.openqa.selenium.remote.DesiredCapabilities;
+
+import com.frameworkium.core.ui.driver.AbstractDriver;
+
+import io.appium.java_client.windows.WindowsDriver;
+import io.appium.java_client.windows.WindowsElement;
+
+public class WinAppDriverImpl extends AbstractDriver {
+
+	protected static final Logger logger = LogManager.getLogger();
```
Comment (ham1) [inline src/main/java/com/frameworkium/core/ui/driver/drivers/WinAppDriverImpl.java]: spaces not tabs please

File: src/test/java/com/frameworkium/integration/angularjs/tests/web/DocumentationTest.java [hunk 5]
```diff
@@ -16,7 +16,7 @@ public void test_configure_browser_before_use() {
         configureBrowserBeforeUse();
         DeveloperGuidePage.open();
     }
-
+
```
Comment (ham1) [inline src/test/java/com/frameworkium/integration/angularjs/tests/web/DocumentationTest.java]: probably shouldn't be in the PR?

**Comment by Hyphen31** [comment]
Changes made as requested

**Pull request closed without merging**